modules/tinymce/src/plugins/table/test/ts/browser/settings/TableCellClassListTest.ts

```diff
@@ -35,8 +40,48 @@ describe('browser.tinymce.plugins.table.TableCellClassListTest', () => {
     editor.setContent(tableHtml);
     TinySelections.setSelection(editor, [ 0, 0, 0, 0, 0 ], 0, [ 0, 0, 0, 0, 0 ], 1);
     editor.execCommand('mceTableCellProps');
-    await TableTestUtils.pAssertListBoxValue('Select class', editor, 'Class', 'test');
+    await TableTestUtils.pAssertListBoxValue('Select class', editor, 'Class', 'mce-no-match');
     await TableTestUtils.pClickDialogButton(editor, true);
-    TinyAssertions.assertContentPresence(editor, { 'td.test': 1 });
+    TinyAssertions.assertContent(editor, tableHtml);
+  });
+
+  it('TINY-6653: Selecting a cell class that has no value should remove a previously applied class', async () => {
+    const editor = hook.editor();
+    editor.options.set('table_cell_class_list', [
+      { title: 'none', value: '' },
+      { title: 'test', value: 'test' }
+    ]);
+    editor.setContent(tableHtml);
+    TinySelections.setSelection(editor, [ 0, 0, 0, 0, 0 ], 0, [ 0, 0, 0, 0, 0 ], 1);
```

Comment:
It probably would be good to assert that the `Select...` list box item is selected when there is no match and the `{ title: 'none', value: '' }` is in the `table_cell_class_list`.

Now that I think about it, could we still run into an issue where the 'None' item is selected for a multi-cell selection?

Reply:
"None" is selected if at least one cell has no class yes

Reply:
I've addressed this now by using `'mce-no-match'` when there's conflicting class values for multi cell/row selections